generators/pkg/generator/generate_test.go

```diff
@@ -103,3 +105,85 @@ func TestGeneratePipeline(t *testing.T) {
 		t.Errorf("Pipeline mismatch (-want +got):\n %s", diff)
 	}
 }
+
+func TestGenerateTrigger(t *testing.T) {
+	// read the want TriggerBinding
+	path := "./testdata/triggerbinding.yaml"
+	file, err := ioutil.ReadFile(path)
+	if err != nil {
+		t.Fatalf("fail to read file %s: %v", path, err)
+	}
+
+	tb := &v1alpha1.TriggerBinding{}
+	if err := yaml.Unmarshal(file, tb); err != nil {
+		t.Fatalf("fail to unmarshal from the input: %v", err)
+	}
```

Comment:
This section of code is also duplicative. We can break this out into a helper func to make this simpler. e.g.

```go
func unmarshal(t *testing.T, path string, i interface{}) {
	t.Helper()

	file, err := ioutil.ReadFile(path)
	if err != nil {
		t.Fatalf("fail to read file %s: %v", path, err)
	}

	if err := yaml.Unmarshal(file, i); err != nil {
		t.Fatalf("fail to unmarshal from the input: %v", err)
	}
}

func Test(t *testing.T) {
  tb := &v1alpha1.TriggerBinding{}
  unmarshal(t, "./testdata/triggerbinding.yaml", tb)
}
...